lua/autorun/server/sv_pvp_movespeed.lua

```diff
@@ -83,6 +96,26 @@ local function getPlayerWeight( ply )
     return totalWeight
 end
 
+local function dropPlyWeapon( ply )
+    local currentWeapon = ply:GetActiveWeapon():GetClass()
+    if isUndroppable[currentWeapon] then
+        ply:ChatPrint("This weapon is unable to be dropped!")
+        return
+    end
+
+    ply:StripWeapon( currentWeapon )
```

Comment:
couldn't you use player:DropWeapon here?

Reply:
Drop weapon doesn't return an entity so we can't despawn the weapon after a given time frame.

Reply:
the weapon entity doesn't change when its dropped